src/rust/mqtt-publisher/src/main.rs

```diff
@@ -0,0 +1,230 @@
+                        ),
+                        QoS::AtLeastOnce,
+                        false,
+                        serde_json::to_string(&data)?,
+                    )
+                    .await?;
+            }
+            "place_market_order" => {
+                let data: PlaceMarketOrderNotif = serde_json::from_str(notification.payload())?;
+                client
+                    .publish(
+                        format!(
+                            "place_market_order/{}/{}/{}/{}",
+                            data.market_id, data.user, data.custodian_id, data.integrator
+                        ),
+                        QoS::AtLeastOnce,
+                        false,
+                        serde_json::to_string(&data)?,
+                    )
+                    .await?;
+            }
+            "place_swap_order" => {
+                let data: PlaceSwapOrderNotif = serde_json::from_str(notification.payload())?;
+                client
+                    .publish(
+                        format!(
+                            "place_swap_order/{}/{}/{}",
+                            data.market_id, data.integrator, data.signing_account
+                        ),
+                        QoS::AtLeastOnce,
+                        false,
+                        serde_json::to_string(&data)?,
+                    )
+                    .await?;
+            }
+            "change_order_size" => {
+                let data: ChangeOrderSizeNotif = serde_json::from_str(notification.payload())?;
+                client
+                    .publish(
+                        format!(
+                            "change_order_size/{}/{}/{}",
+                            data.market_id, data.user, data.custodian_id
+                        ),
+                        QoS::AtLeastOnce,
+                        false,
+                        serde_json::to_string(&data)?,
+                    )
+                    .await?;
+            }
+            "cancel_order" => {
+                let data: CancelOrderNotif = serde_json::from_str(notification.payload())?;
+                client
+                    .publish(
+                        format!(
+                            "cancel_order_notif/{}/{}/{}",
+                            data.market_id, data.user, data.custodian_id
+                        ),
+                        QoS::AtLeastOnce,
+                        false,
+                        serde_json::to_string(&data)?,
+                    )
+                    .await?;
+            }
+            "fill" => {
+                let data: FillNotif = serde_json::from_str(notification.payload())?;
+                client
+                    .publish(
+                        format!(
+                            "fill_notif/{}/{}/{}",
+                            data.market_id, data.maker_address, data.maker_custodian_id
```

Comment:
So here this should also be filterable on `taker` and `taker_custodian_id` or whatever the fields are in SQL

(For example I'm consulting the Move struct and I see that it says `maker` whereas above you've written `maker_address`)